webapp/packages/core-root/src/SessionResource.ts

```diff
@@ -73,6 +58,27 @@ export class SessionResource extends CachedDataResource<SessionState | null> {
     this.defaultLocale = defaultLocale;
   }
 
+  private handleSessionStateEvent(event: ISessionStateEvent) {
+    if (!this.data) {
+      return;
+    }
+
+    const sessionState: SessionState = {
+      ...this.data,
+      valid: event?.isValid ?? this.data?.valid,
+      remainingTime: event.remainingTime,
+      actionParameters: event.actionParameters,
+      cacheExpired: event?.isCacheExpired ?? this.data.cacheExpired,
+      lastAccessTime: String(event.lastAccessTime),
+      locale: event.locale,
+    };
+
+    this.performUpdate(undefined, [], async () => {
+      this.setData(sessionState);
+      this.onStatusUpdate.execute(sessionState);
```

Comment:
performUpdate is asynchronous function and `this.data` may change before update will be executed, please move const here.
Also you can remove `onStatusUpdate` and use `onDataUpdate` instead (it will be called by `performUpdate`)

there also one note, that `performUpdate` probably will trigger to almost all data reload